packages/frontend2/src/app/_components/markdown/glossary-context.tsx

```diff
@@ -0,0 +1,31 @@
+'use client'
+
+import { createContext, useContext } from 'react'
+import { type CollectionEntry } from '~/content/get-collection'
+
+type GlossaryContextValue = {
+  terms: CollectionEntry<'glossary'>[]
+}
+
+const GlossaryContext = createContext<GlossaryContextValue | null>(null)
+
+interface Props {
+  value: GlossaryContextValue
+  children: React.ReactNode
+}
+
+export function GlossaryContextProvider({ children, value }: Props) {
```

Comment:
Since this is public-facing API, consider accepting `terms` as the parameters and not value object.